pkg/chunk/aws/dynamodb_storage_client.go

```diff
@@ -792,17 +746,17 @@ func (b dynamoDBReadRequest) TakeReqs(from dynamoDBReadRequest, max int) {
 	}
 }
 
-func withErrorHandler(tableName, operation string) func(req *request.Request) {
+func withErrorHandler(tableName, operation string, dynamoFailures *prometheus.CounterVec) func(req *request.Request) {
```

Comment:
You may consider to pass the entire `*metrics`. I would be less error prone, otherwise you may do a mistake and pass the wrong metric.